File: lua/001/main.lua
--[[
    ProjectEuler 001
    Lua / main.lua
    copyright (c) 2015 Susisu

    Lua 5.1.5
    `lua main.lua`
]]

function main()
    local tot = 0
    for i = 1, 999 do
        if i % 3 == 0 or i % 5 == 0 then
            tot = tot + i
        end
    end
    print(tot)
end

main()

File: rust/001/main.rs
/*
    ProjectEuler 001
    Rust / main.rs
    copyright (c) 2015 Susisu

    Rust 1.3.0
    `rustc main.rs -o bin/main`
*/

fn main() {
    let n: i32 = (1..1000)
        .filter(|&i| i % 3 == 0 || i % 5 == 0)
        .fold(0, |acc, i| acc + i);
    println!("{}", n);
}
